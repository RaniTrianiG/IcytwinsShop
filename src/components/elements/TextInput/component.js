import React from 'react';
import { Input as TextInput } from 'react-native-elements';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import style from './styles';

const Component = ({
  label,
  meta: { touched, error },
  input: { onChange, placeholder, ...restInput },
  ...rest
}) => {
  return (
    <View style={style.container}>
      <Text style={style.label}>{label}</Text>
      <TextInput
        inputContainerStyle={style.inputContainer}
        errorMessage={error && touched ? error : null}
        errorStyle={style.errorMessage}
        // leftIcon={<Icon name="user" size={24} color="black" />}
        onChangeText={onChange}
        {...restInput}
        {...rest}
      />
    </View>
  );
};

Component.propTypes = {
  label: PropTypes.string.isRequired,
  meta: PropTypes.object.isRequired,
  input: PropTypes.object.isRequired
};

export default Component;
