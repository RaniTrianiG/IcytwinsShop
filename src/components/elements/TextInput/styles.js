import { StyleSheet } from 'react-native';

const style = StyleSheet.create({
  container: {
    marginBottom: 5
  },
  errorMessage: {
    color: 'red',
    fontSize: 11
  },
  inputContainer: {
    borderColor: '#ccc',
    borderRadius: 5,
    borderWidth: 1,
    marginBottom: 5,
    padding: 3
  },
  label: {
    marginBottom: 10,
    marginLeft: 10
  }
});

export default style;
