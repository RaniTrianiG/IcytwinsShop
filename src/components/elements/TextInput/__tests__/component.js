import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import TextInput from '../component';
// Note: test renderer must be required after react-native.

it('renders correctly', () => {
  const props = {
    meta: { touched: false, error: null },
    input: { onChange: jest.fn(), placeholder: 'placeholder' },
    label: 'some Label'
  };
  const wrapper = shallow(<TextInput {...props} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly after touched and get some error', () => {
  const props = {
    meta: { touched: true, error: 'something wrong' },
    input: { onChange: jest.fn(), placeholder: 'placeholder' },
    label: 'some Label'
  };
  const wrapper = shallow(<TextInput {...props} />);
  expect(wrapper).toMatchSnapshot();
});
