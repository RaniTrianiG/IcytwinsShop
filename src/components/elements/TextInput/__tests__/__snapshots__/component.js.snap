// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
    }
  }
>
  <Text
    style={
      Object {
        "marginBottom": 10,
        "marginLeft": 10,
      }
    }
  >
    some Label
  </Text>
  <ForwardRef(Themed.Input)
    errorMessage={null}
    errorStyle={
      Object {
        "color": "red",
        "fontSize": 11,
      }
    }
    inputContainerStyle={
      Object {
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": 5,
        "padding": 3,
      }
    }
    onChangeText={[MockFunction]}
  />
</View>
`;

exports[`renders correctly after touched and get some error 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
    }
  }
>
  <Text
    style={
      Object {
        "marginBottom": 10,
        "marginLeft": 10,
      }
    }
  >
    some Label
  </Text>
  <ForwardRef(Themed.Input)
    errorMessage="something wrong"
    errorStyle={
      Object {
        "color": "red",
        "fontSize": 11,
      }
    }
    inputContainerStyle={
      Object {
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": 5,
        "padding": 3,
      }
    }
    onChangeText={[MockFunction]}
  />
</View>
`;
